libs/ddhub-client-gateway-frontend/ui/channels/src/lib/containers/modals/Create/Restrictions/effects/didRestriction.effects.ts

```diff
@@ -35,17 +36,32 @@ export const useDIDRestrictionEffects = (currentDids: string[]) => {
   };
 
   const addDID = (did: string) => {
-    if (dids.includes(did)) {
-      return;
+    if (!dids.includes(did)) {
+      setDids([did, ...dids]);
     }
-    setDids([...dids, did]);
+
     clearDIDInput();
   };
 
   const removeDID = (value: string) => {
     setDids(dids.filter((did) => did !== value));
   };
 
+  const updateDID = (type: RestrictionType, oldValue: string, newValue: string) => {
+    const filteredDids = dids.filter((did) => did !== oldValue);
+
+    if (!filteredDids.includes(newValue)) {
+      const didsToSet = [...filteredDids];
```

Comment:
filter is already creating new list -> no reference so i think there is no need to create new list.